test/support/lib/NockHelper.ts

```diff
@@ -0,0 +1,86 @@
+// NOTE: This is in the lib folder, no OneSignal specific code should be added here
+
+import nock, { Interceptor } from "nock";
+
+type NockResultFunction = (result: NockRequestResult) => void;
+
+export interface NockRequest {
+  body: any;
+  url: string;
+}
+
+export interface NockResponse {
+  status: number;
+  body: any;
+}
+
+export interface NockRequestResult {
+  request: NockRequest;
+  response: NockResponse;
+}
+
+export interface NockScopeWithResultPromise {
+  nockScope: nock.Scope;
+  result: Promise<NockRequestResult>;
+}
+
+type NockInterceptMethods = "post" | "put" | "get";
+
+interface NockOptions {
+  reply: NockResponse;
+  baseUrl: string;
+  path: string;
+  method: NockInterceptMethods;
+}
+
+export class NockHelper {
```

Comment:
Deciding that this can be a comment in the file instead of when and why you should use this NockHelper. More details on how tests should be written should be done later in another PR>

Reply:
Done, resolve if the comment I added is clear.